tests/test_common.py

```diff
@@ -0,0 +1,251 @@
+        with pytest.raises(NotImplementedError):
+            genie_class.from_bytes(ByteHandler(b''))
+
+    def test_from_bytes_with_count(self, genie_class: GenieClass):
+        with pytest.raises(NotImplementedError):
+            genie_class.from_bytes_with_count(ByteHandler(b''), 0)
+
+    def test_to_bytes(self, genie_class: GenieClass):
+        with pytest.raises(NotImplementedError):
+            genie_class.to_bytes(Version.UNDEFINED)
+
+    def test_write_debug_string(self, genie_class: GenieClass):
+        assert genie_class.write_debug_string("foobar") == b'`\n\x06\x00foobar'
+
+    def test_write_string(self, genie_class: GenieClass):
+        assert genie_class.write_string(6, "foobar") == b'foobar'
+
+    def test_write_int_8(self, genie_class: GenieClass):
+        assert genie_class.write_int_8(10) == b'\n'
+        with pytest.raises(OverflowError):
+            genie_class.write_int_8(1000)
+
+    def test_write_int_8_array(self, genie_class: GenieClass):
+        assert genie_class.write_int_8_array([10, 10, 10]) == b'\n\n\n'
+        with pytest.raises(OverflowError):
+            genie_class.write_int_8_array([10, 10, 1000])
+
+    def test_write_int_16(self, genie_class: GenieClass):
+        assert genie_class.write_int_16(10) == b'\n\x00'
+        with pytest.raises(OverflowError):
+            genie_class.write_int_16(70000)
+
+    def test_write_int_16_array(self, genie_class: GenieClass):
+        assert genie_class.write_int_16_array([10, 10, 10]) == b'\n\x00\n\x00\n\x00'
+        with pytest.raises(OverflowError):
+            genie_class.write_int_16_array([10, 10, 70000])
+
+    def test_write_int_32(self, genie_class: GenieClass):
+        assert genie_class.write_int_32(10) == b'\n\x00\x00\x00'
+        with pytest.raises(OverflowError):
+            genie_class.write_int_32(5000000000)
+
+    def test_write_int_32_array(self, genie_class: GenieClass):
+        assert genie_class.write_int_32_array([10, 10, 10]) == b'\n\x00\x00\x00\n\x00\x00\x00\n\x00\x00\x00'
+        with pytest.raises(OverflowError):
+            genie_class.write_int_32_array([10, 10, 5000000000])
+
+    def test_write_float(self, genie_class: GenieClass):
+        assert genie_class.write_float(10) == b'\x00\x00 A'
+
+    def test_write_float_array(self, genie_class: GenieClass):
+        assert genie_class.write_float_array([10, 10, 10]) == b'\x00\x00 A\x00\x00 A\x00\x00 A'
+
+    def test_write_class(self, genie_class: GenieClass):
+        with pytest.raises(NotImplementedError):
+            genie_class.write_class(GenieClass(), Version.UNDEFINED)
+
+    def test_write_class_array(self, genie_class: GenieClass):
+        with pytest.raises(NotImplementedError):
+            genie_class.write_class(GenieClass(), Version.UNDEFINED)
+
+
+class TestByteHandler:
+    @pytest.fixture
+    def byte_handler(self):
+        data = memoryview(b''.join([int.to_bytes(i) for i in range(256)]))
```

Comment:
Using this data makes all of the resulting numbers (except in the `read_int_8_…` tests) very hard to understand/make sense of the numbers. Maybe ditch the fixture for setting up sensible data in each test?

Reply:
I don't disagree with this sentiment. Admittedly this is my first go with `pytest` (I've typically used `unittest` for all my projects) and I was mostly trying to get a feel for it and how fixtures work.

Let me see if I can put together something more in-line with what you're looking for

Reply:
Alright, I kept the fixtures, but gave them easier to read values. I wanted to use an actual `.dat` file for these tests, but given how they're at the absolute core of the library it seemed best to use dummy data. Further tests are going to use proper `.dat` files where possible